patroni/dcs/etcd3.py

```diff
@@ -908,8 +909,8 @@ def set_history_value(self, value: str) -> bool:
         return bool(self._client.put(self.history_path, value))
 
     @catch_etcd_errors
-    def set_sync_state_value(self, value: str, index: Optional[str] = None) -> bool:
-        return self.retry(self._client.put, self.sync_path, value, mod_revision=index)
+    def set_sync_state_value(self, value: str, index: Optional[str] = None) -> Union[str, bool]:
+        return self.retry(self._client.put, self.sync_path, value, mod_revision=index).get('header', {}).get('revision')
```

Comment:
Can it also return None? Shouldn't it be `.get('header', {}).get('revision', False)`

Reply:
```suggestion
        return self.retry(self._client.put, self.sync_path, value, mod_revision=index)\
            .get('header', {}).get('revision', False)
```